### ISSUE TITLE
[1.9.3] Profile pictures are not displayed in native notifications on GNOME

### ISSUE THREAD

**Issue by Bryophyllum** [flathub/org.telegram.desktop#122]
With the 1.9.3 update profile pictures are not present in system-native notifications on GNOME 3.34:
![nopfpnotification](https://user-images.githubusercontent.com/27376783/71939292-7a0f8a80-31b2-11ea-8f6d-8e75d4b43acf.png)

Reverting to an older commit and downgrading from the commit `b36dc8e9b332272674a19fcf7ffe2de7d5cfae0a0e6aa66375f1544082930d74` by issuing the command: 
```
$ sudo flatpak update \
--commit=6a97909897328564f0540b62e4dc7537ed021a2e7714b525e50c9ef3acb28749 org.telegram.desktop
```
..brings them back.

```
$ flatpak remote-info --log flathub_beta org.telegram.desktop

Telegram Desktop - Telegram Desktop messenger

        ID: org.telegram.desktop
       Ref: app/org.telegram.desktop/x86_64/beta
      Arch: x86_64
    Branch: beta
   Version: 1.8.15
   License: GPL-3.0
Collection: org.flathub.Beta
  Download: 23.2 MB
 Installed: 49.6 MB
   Runtime: org.kde.Platform/x86_64/5.13

       Sdk: org.kde.Sdk/x86_64/5.13
    Commit: b36dc8e9b332272674a19fcf7ffe2de7d5cfae0a0e6aa66375f1544082930d74
    Parent: 6a97909897328564f0540b62e4dc7537ed021a2e7714b525e50c9ef3acb28749
   Subject: Redo font dirs patch (b46bbbdf)
      Date: 2020-01-07 13:37:36 +0000
   History: 

    Commit: 6a97909897328564f0540b62e4dc7537ed021a2e7714b525e50c9ef3acb28749
   Subject: Drop bundled ffmpeg and use .ffmpeg-full extension (5baa2dbf)
      Date: 2020-01-07 07:34:26 +0000
```

**Comment by grulja** [comment]
I wonder why this might be happenning. @gasinvein any idea?

**Comment by gasinvein** [comment]
Maybe it's upstream bug (or intended design change)? AFAIK Telegram was changing something with notification system to use dbus/qdbus directly and avoid depending on libnotify.

**Comment by xvitaly** [comment]
On Fedora/KDE with RPM build notifications works fine with icons.

**Comment by gasinvein** [comment]
Seems like this is specific to host DE: tried VM with KDE and this flatpak - profile pictures are shown there.

**Comment by ilya-fedin** [comment]
Can you give results of these commands?
```
busctl --user call org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications GetServerInformation
busctl --user call org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications GetCapabilities
```

**Comment by gasinvein** [comment]
@ilya-fedin  Via dbus-send (busctl unavailable in flatpak sandbox):
```
   string "gnome-shell"
   string "GNOME"
   string "3.34.3"
   string "1.2"
```
```
   array [
      string "actions"
      string "body"
      string "body-markup"
      string "icon-static"
      string "persistence"
      string "sound"
   ]
```

**Comment by ilya-fedin** [comment]
@gasinvein okay, can you find the "Notification daemon specification version" string in telegram log and check that it is the same (1.2)?
If it is the same, it is will be very helpful if you install bustle (it is in flatpak) and check that Telegram sends notification (on session bus) with `image-data` property and not `image_data` or `icon_data`

**Comment by gasinvein** [comment]
@ilya-fedin Sorry, what's telegram log and how do I enable/get it?

**Comment by ilya-fedin** [comment]
@gasinvein non-flatpak version has it in ~/.local/share/TelegramDesktop/log.txt
About flatpak version I don't remember

**Comment by gasinvein** [comment]
So, yes, log shows the same version/capabilities:
```
[2020.01.15 13:11:06] Notification daemon specification version: 1.2
[2020.01.15 13:11:06] Notification daemon capabilities: actions, body, body-markup, icon-static, persistence, sound
```
In bustle, it looks like telegram calls `org.freedesktop.Notifications.Notify` with `image-data` property indeed.

**Comment by ilya-fedin** [comment]
> In bustle, it looks like telegram calls `org.freedesktop.Notifications.Notify` with `image-data` property indeed.

This is very strange, everything should work

**Comment by ilya-fedin** [comment]
By the way, you can compare dbus messages on old version and the new and look for any differences

**Comment by gasinvein** [comment]
Arguments Telegram sends to `.Notify()` are the following:
Telegram 1.9:
```
@s "Telegram Desktop"
@u 0
@s "telegram"
@s "<bytes>"
@s "<bytes>"
@as ["mail-reply-sender", "<bytes>"]
@a{sv} {"category": <@s "im.received">, "desktop-entry": <@s "telegramdesktop">, "image-data": <@(iiibiiay) (80, 80, 320, True, 8, 4, [<array of bytes>])>, "suppress-sound": <@b True>}
```
Telegram 1.8:
```
@s "Telegram Desktop"
@u 0
@s ""
@s "<bytes>"
@s "<bytes>"
@as ["default", "Reply"]
@a{sv} {"desktop-entry": <@s "telegramdesktop">, "image-data": <@(iiibiiay) (80, 80, 320, True, 8, 4, [<array of bytes>])>}
```

**Comment by ilya-fedin** [comment]
Everything looks ok, but I have an idea :)
Try `busctl --user call org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications Notify 'susssasa{sv}i' "" 0 "" "Test" "" 0 1 category s im.received \ -1` (I think you can execute it on the host)

**Comment by gasinvein** [comment]
>Try `busctl --user call org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications Notify 'susssasa{sv}i' "" 0 "" "Test" "" 0 1 category s im.received \ -1` (I think you can execute it on the host)

It shows notification with default icon.

**Comment by ilya-fedin** [comment]
> It shows notification with default icon.

I never used GNOME3 and I don't understand what this means

**Comment by gasinvein** [comment]
Well, there is (i) symbol on the notification. I believe it's the fallback when no other icon is set.

**Comment by ilya-fedin** [comment]
Okay, I found the root of the problem, I'll pr fix to upstream

**Comment by gasinvein** [comment]
Will be fixed by #126

**Comment by Bryophyllum** [comment]
Indeed. I'm closing the issue. Thanks to everyone involved.
